src/kernels/batchnorm_functions.h

```diff
@@ -54,6 +59,31 @@
 
 #endif
 
+#if MIOPEN_USE_BFPMIX == 1
+#define _FLOAT ushort
```

Comment:
It must not be `ushort`, `ushort` is just a container type holding 2 unsigned bytes.
Casting to that type is dangerous.

Reply:
I think that's what we want in this case.
Any where casting or doing math to that type is wrong.

Reply:
yes ushort is what we want.  it is passed in as ushort and nothing should nakedly be using _FLOAT for any math operations or assuming it can naturally convert to _FLOAT_ACCUM, _FLOAT_PREC, or even just float/int/...  any use of an argument or variable that is _FLOAT need to use the FLOAT2ACCUM or FLOAT2FLOATPREC and this gives you back a value you can actually do math with.